shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -0,0 +1,217 @@
+                                                   int id,
+                                                   int action,
+                                                   ByteBuffer args,
+                                                   int argsPosition);
+
+  public native void nativeSetSemanticsEnabled(long nativePlatformViewAndroid, boolean enabled);
+
+  public native void nativeSetAccessibilityFeatures(long nativePlatformViewAndroid, int flags);
+
+  public native boolean nativeGetIsSoftwareRenderingEnabled();
+
+  public native void nativeRegisterTexture(long nativePlatformViewAndroid, long textureId, SurfaceTexture surfaceTexture);
+
+  public native void nativeMarkTextureFrameAvailable(long nativePlatformViewAndroid, long textureId);
+
+  public native void nativeUnregisterTexture(long nativePlatformViewAndroid, long textureId);
+  //------- End from FlutterView -----
+
+  // TODO(mattcarroll): rename comments after refactor is done and their origin no longer matters
+  //------ Start from FlutterNativeView ----
+  public native long nativeAttach(FlutterJNI flutterJNI, boolean isBackgroundView);
+  public native void nativeDestroy(long nativePlatformViewAndroid);
+  public native void nativeDetach(long nativePlatformViewAndroid);
+
+  public native void nativeRunBundleAndSnapshotFromLibrary(
+      long nativePlatformViewAndroid,
+      @NonNull String pathToBundleWithEntrypoint,
+      @Nullable String pathToFallbackBundle,
+      @Nullable String entrypointFunctionName,
+      @Nullable String pathToEntrypointFunction,
+      @NonNull AssetManager manager
+  );
+
+  public native String nativeGetObservatoryUri();
+
+  // Send an empty platform message to Dart.
+  public native void nativeDispatchEmptyPlatformMessage(
+      long nativePlatformViewAndroid,
+      String channel,
+      int responseId
+  );
+
+  // Send a data-carrying platform message to Dart.
+  public native void nativeDispatchPlatformMessage(
+      long nativePlatformViewAndroid,
+      String channel,
+      ByteBuffer message,
+      int position,
+      int responseId
+  );
+
+  // Send an empty response to a platform message received from Dart.
+  public native void nativeInvokePlatformMessageEmptyResponseCallback(
+      long nativePlatformViewAndroid,
+      int responseId
+  );
+
+  // Send a data-carrying response to a platform message received from Dart.
+  public native void nativeInvokePlatformMessageResponseCallback(
+      long nativePlatformViewAndroid,
+      int responseId,
+      ByteBuffer message,
+      int position
+  );
+  //------ End from FlutterNativeView ----
+
+  // TODO(mattcarroll): rename comments after refactor is done and their origin no longer matters
+  //------ Start from Engine ---
+  @SuppressWarnings("unused")
+  private void onPreEngineRestart() {
+    for (EngineLifecycleListener listener : engineLifecycleListeners) {
+      listener.onPreEngineRestart();
+    }
+  }
+  //------ End from Engine ---
+}
```

Comment:
uber nit: lets be consistent about newlines at end of file

Reply:
Ok, I'll remove the trailing newline from the one file that has it (FlutterEngine.java) and push the change.